### ISSUE TITLE
Clicks with multiple segments 

### ISSUE THREAD

**Issue by jacopogrecodalceo** [johannphilippe/hypercurve#10]
Hello again,

Maybe i'm trying to push the boundaries, but I cannot figure out what's causing clicks with this curves. 
The interesting part is that the less length, the less the clicks.

`girich	hc_gen 0, 8192, 0, \ 
		hc_segment(1/40, 0.19607, hc_blackman_curve()), \ 
		hc_segment(1/40, 0.65294, hc_kiss_curve()), \ 
		hc_segment(1/40, 0.29282, hc_toxoid_curve(2.6382)), \ 
		hc_segment(1/40, 0.15045, hc_hanning_curve()), \ 
		hc_segment(2/40, 0.77689, hc_gaussian_curve(0.44594, 0.57684)), \ 
		hc_segment(5/40, 0.30218, hc_cubic_curve()), \ 
		hc_segment(2/40, 0.25578, hc_hanning_curve()), \ 
		hc_segment(1/40, 1, hc_toxoid_curve(1.23829)), \ 
		hc_segment(26/40, 0, hc_power_curve(0.43519))

	instr 1

ktrig metro randomi:k(1, 3, .5)

if ktrig == 1 then
	schedulek 2, 0, 1
endif

	endin

	instr 2

idur init p3

aout = oscili:a(1/4, random:i(300, 200))*table3:a(linseg:a(0, idur, 1), girich, 1)
outall aout

	endin`

**Comment by johannphilippe** [comment]
I experienced clicks too today. I could solve it with multiplying the index by 0.99, which seems to me like the guard point is missing (a 0 at the end of the curve). 
Can you say if it was similar for you before ? Did you change anything recently ?

**Comment by jacopogrecodalceo** [comment]
I did not change anything, but I feel it depends also of the duration of the sound.
I think that some curves are so steep that they become clicks even with interpolation if they're read quickly. Could it be possible?

To me if I multiply *.99 does not change anything !

**Comment by johannphilippe** [comment]
Yes, I also experienced some clicks related to the fact that some segments were very short, or with too much slope (or both) - particularly with `tightrope_walker_curve`. I will check this when I have time !
